corehq/apps/app_manager/views/settings.py

```diff
@@ -73,6 +74,12 @@ def edit_commcare_profile(request, domain, app_id):
                 app.profile[settings_type] = {}
             app.profile[settings_type][name] = value
             changed[settings_type][name] = value
+
+    if not domain_has_privilege(domain, privileges.APP_DEPENDENCIES):
+        # remove dependencies if they were set before
+        if 'dependencies' in app.profile.get('features', {}):
```

Comment:
Non blocking nit: because we're doing the same in [this commit](https://github.com/dimagi/commcare-hq/pull/35237/commits/791ada683322e0253e2190221a065d728873bec8), maybe create a separate function for doing this?

Reply:
Oh yes that is the same thing at both places.

I believe I would be extracting just this bit
```
if 'dependencies' in app.profile.get('features', {}):
    del app.profile['features']['dependencies']
```

That seems a bit overkill?

Reply:
Although it's only 2 lines, it's the concept that I was thinking of extracting, but I'll leave it to you to decide. I don't have strong feelings either way.

Reply:
I agree with the concept but going to leave this as it is here.